tests/gha-run-k8s-common.sh

```diff
@@ -161,7 +161,7 @@ function deploy_k3s() {
 	if [ "${ARCH}" = "x86_64" ]; then
 		ARCH=amd64
 	fi
-	kubectl_version=$(/usr/local/bin/k3s kubectl version --short 2>/dev/null | grep "Client Version" | sed -e 's/Client Version: //' -e 's/\+k3s1//')
+	kubectl_version=$(/usr/local/bin/k3s kubectl version --short 2>/dev/null | grep "Client Version" | sed -e 's/Client Version: //' -e 's/\+k3s[0-9]\+//')
```

Comment:
It seems that we want to match several trailing digits, not a plus sign:

```suggestion
	kubectl_version=$(/usr/local/bin/k3s kubectl version --short 2>/dev/null | grep "Client Version" | sed -e 's/Client Version: //' -e 's/\+k3s[0-9]+//')
```

Reply:
Oops I was wrong, we need to escape the final plus sign. The one before k3s doesn't need to be escaped though.

Reply:
As discussed on Slack, we actually have to escape the latter +, but the former one is unnecessarily escaped right now.
I've added a commit to address that.